src/filebrowser_formaction.cpp

```diff
@@ -261,8 +261,8 @@ char filebrowser_formaction::get_filetype(mode_t mode) {
 		{ S_IFIFO, 'p' }, { S_IFLNK, 'l' }, { 0      ,  0  }
 	};
 	for (unsigned int i=0; flags[i].flag != 0; i++) {
-		if (mode & flags[i].flag)
-			return flags[i].ftype;
+          if ((mode & S_IFMT) == flags[i].flag)
```

Comment:
Please use tabs for indentation. [Editorconfig plugin](http://editorconfig.org/) will take care of it if you install it.